lib/jsdom/living/constraint-validation/DefaultConstraintValidation-impl.js

```diff
@@ -0,0 +1,60 @@
+"use strict";
+
+const ValidityState = require("../generated/ValidityState");
+const Event = require("../generated/Event");
+
+exports.implementation = class DefaultConstraintValidationImpl {
+  get _customValidityErrorMessage() {
+    if (typeof this._customErrorMessage !== "string") {
+      return "";
+    }
+    return this._customErrorMessage;
+  }
+
+  // https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-willvalidate
+  get willValidate() {
+    return false;
+  }
+
+  get validity() {
+    if (!this._validity) {
+      this._validity = ValidityState.createImpl(this);
+    }
+    return this._validity;
+  }
+
+  // https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-checkvalidity
+  checkValidity() {
+    if (!this.willValidate) {
+      return true;
+    }
+    const { validity } = this;
+    if (!validity.valid) {
+      this.dispatchEvent(Event.createImpl(["invalid", { cancelable: true }]));
+    }
+    return validity.valid;
+  }
+
+  // https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-setcustomvalidity
+  setCustomValidity(message) {
+    this._customErrorMessage = message;
+  }
+
+  // https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-reportvalidity
+  // Since jsdom has no user interaction, it's the same as #checkValidity
+  reportValidity() {
+    return this.checkValidity();
+  }
+
+  // https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-validationmessage
+  get validationMessage() {
+    const { validity } = this;
+    if (!this.willValidate || validity.valid) {
+      return "";
+    }
+    if (validity.customError) {
+      return this._customErrorMessage;
+    }
```

Comment:
This is another area where, since you are using the public API, it is hard to tell if you are following the spec.

Reply:
To follow the spec would mean to put `this._customValidityErrorMessage = ''` in every constructor of elements that support validation (or alternatively have an `initialiseCustomValidityErorrMessage/init` method on the default constraint api mixin and call that from the constructor), let me know which way you prefer.

Reply:
I went with this solution for now - `this._customValidityErrorMessage` set to '' in the constructor for elements that are not barred from constraint validation